compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java

```diff
@@ -1333,4 +624,14 @@ private void acceptNode(BLangNode node, SymbolEnv env) {
         node.accept(this);
         this.symbolEnv = prevEnv;
     }
+
+    private boolean isNarrowerEnclosure(Location nodePosition) {
+        // Have to special case the pkg node since its position is (0,0,0,0). Plus any other node would for sure will
+        // be a narrower enclosure than the package.
+        if (this.scope == null || this.scope.node.getKind() == NodeKind.PACKAGE) {
+            return true;
+        }
+
+        return PositionUtil.withinRange(nodePosition.lineRange(), this.scope.node.getPosition());
+    }
```

Comment:
Seems like this is called with `PositionUtil.withInBlock` together many times, What if we add this inside `PositionUtil.withInBlock()` method ?

Reply:
This check is specific to the env resolver though. If there's a requirement from other components as well for this, can consider moving it to `PositionUtil`.